ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/InMemory.hs

```diff
@@ -13,6 +13,7 @@
 {-# LANGUAGE ScopedTypeVariables    #-}
 {-# LANGUAGE TypeApplications       #-}
 {-# LANGUAGE TypeFamilies           #-}
+{-# LANGUAGE UndecidableInstances   #-}
```

Comment:
Hmm, where is this coming from? The migration notes for 8.10 say this might be needed where we have some injective type families, but I don't see that here...

Reply:
This is the error I get without it:
```
src/Ouroboros/Consensus/Storage/LedgerDB/InMemory.hs:311:3: error:
    • Variable ‘c'’ occurs more often
        in the constraint ‘c'’ than in the instance head ‘c’
      (Use UndecidableInstances to permit this)
    • In the quantified constraint ‘c' => c’
      In the definition of data constructor ‘Weaken’
      In the data type declaration for ‘Ap’
    |
311 |   Weaken :: (c' => c) => Ap m l r b c -> Ap m l r b c'
    |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Reply:
It's kind of strange, because it's a GADT constructor, not an instance declaration (probably due to GHC's error reporting).